### PR TITLE
Backport PR when label is set

### PR DESCRIPTION
**Pull request by Madhu-1**
backport PR to release-v1.2.0 branch from master branch  when a label is set on PR

Signed-off-by: Madhu Rajanna <madhupr007@gmail.com>

### CODE REVIEW

**Comment by Madhu-1** [comment]
This PR needs to be merged manually as there  is a change in mergify rules

File: .mergify.yml [hunk 1]
```diff
@@ -10,12 +10,20 @@ pull_request_rules:
   - name: automatic merge
     conditions:
       - label!=DNM
-      - '#approved-reviews-by>=2'
-      - 'status-success=continuous-integration/travis-ci/pr'
+      - "#approved-reviews-by>=2"
+      - "status-success=continuous-integration/travis-ci/pr"
```
Comment (nixpanic) [inline .mergify.yml]: Why change the quoting?

File: .mergify.yml [hunk 2]
```diff
@@ -10,12 +10,20 @@ pull_request_rules:
...
       merge:
         method: rebase
         rebase_fallback: merge
         strict: smart
       dismiss_reviews: {}
       delete_head_branch: {}
+  - name: backport patches to release branch
+    conditions:
+      - base=master
+      - label=backport-to-release-v1.2.0
```
Comment (nixpanic) [inline .mergify.yml]: There should be a document with the meanings of the labels. This change should add the automatic backport feature to that doc. But I can not find the document... Maybe `./docs/development-guide.md` would be the right place?

Comment (Madhu-1) [reply .mergify.yml]: updated doc

File: .mergify.yml (hunk 1 above)
Comment (Madhu-1) [reply .mergify.yml]: done by the editor, mergify rules are verified its correct

Comment (nixpanic) [reply .mergify.yml]: Ok.

**Nit:** It is better to make these changes as a separate commit, or mention is in the commit message as not being related to the functional change. What should be done when someone else uses a different editor/settings and the quoting gets replaced again?

File: .mergify.yml (hunk 2 above)
Comment (nixpanic) [reply .mergify.yml]: Thanks!

File: docs/development-guide.md [hunk 3]
```diff
@@ -89,3 +89,10 @@ need to be met before it will be merged:
 
 When the criteria are met, a project maintainer can merge your changes into
 the project's master branch.
+
+### Backporting a PR to release branch
+
+If the PR needs to be backport to a release branch, add
+`backport-to-release-vX.Y.Z` label to the PR. Mergify bot will take care of
```
Comment (nixpanic) [inline docs/development-guide.md]: How can contributors add the label? I can not do so...

Comment (Madhu-1) [reply docs/development-guide.md]: updated doc, once maintainers can add  labels

File: .mergify.yml (hunk 1 above)
Comment (Madhu-1) [reply .mergify.yml]: good  question removed the change

**Pull request merged**